starlette/datastructures.py

```diff
@@ -215,6 +231,14 @@ def __init__(self, value: typing.Union[str, typing.Sequence[str]]):
     def __len__(self) -> int:
         return len(self._items)
 
+    @typing.overload
+    def __getitem__(self, index: int) -> str:
+        ...  # pragma: no cover
+
+    @typing.overload
+    def __getitem__(self, index: slice) -> typing.Sequence[str]:
+        ...  # pragma: no cover
```

Comment:
If this PR doesn't get merged, you can also create a separate PR for this. :+1:

Reply:
Yep may have to split this up